gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java

```diff
@@ -298,6 +300,7 @@ protected void doStart() throws Exception {
                 .handler(new LoginSocialAuthenticationHandler(identityProviderManager, jwtService, certificateManager))
                 .handler(policyChainHandler.create(ExtensionPoint.PRE_LOGIN))
                 .handler(new LoginHideFormHandler(domain))
```

Comment:
The HideLogin form redirect to the external IDP only if there is only one External IDP configured. Otherwise, the login page is displayed without the user/password form and show only the external IDP. So in this condtion, executing the LoginSelectionRuleHandler before or after has no impact on the final behaviour.